config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get "users/:user_id/posts", to: "posts#index"
+  get "users/:user_id/posts/:id", to: "posts#show"
+  get "users", to: "users#index"
+  get "users/:id", to: "users#show"
   # Defines the root path route ("/")
```

Comment:
- [ ] **[OPTIONAL]** We can define these routes more efficiently using the `resources` helper as provided by rails.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```